### PR TITLE
Chore: Add Dependabot config and safe merge workflow

### PR DESCRIPTION
**Pull request by mattrace-gloo**
## 🔥 TLDR
Adds Dependabot configuration and a low-risk merge workflow (Cursor rule + docs) adapted for Fluent Mobile's bare React Native 0.84 stack with npm and Android CI.

## 📋 Summary
- **Issue**: No automated dependency update process or guardrails for safely merging Dependabot PRs in fluent-mobile.
- **Root Cause**: Dependabot was not configured; no repo-specific process existed for validating and sequencing dependency bumps on this codebase.
- **Solution**: Dependabot setup: `.github/dependabot.yml`, `.cursor/rules/dependabot-workflow.mdc`, process guide, resolution log template, and onboarding pointers.
- **Impact**: Weekly grouped npm + GitHub Actions PRs with RN 0.84 / React pin ignores, one-at-a-time lockfile merge policy, and CI-aligned validation steps for agents and humans.

**Related Issue:** N/A (infrastructure chore)

**Type of Change:**
- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [x] 🔧 Maintenance/refactor (no functional changes)

---

### Key Files Modified

**`.github/dependabot.yml`**
- Weekly npm + GitHub Actions updates (Monday 09:00)
- Ignores: `react-native` / `@react-native/*` >=0.85, React/react-test-renderer major+minor bumps, `@op-engineering/op-sqlite` major bumps
- Groups: react-native, react, navigation, testing, dev-tools, tanstack
- Reviewer: `teamgloo/mobile-team` (confirm correct for this repo)
- Limits: 5 open npm PRs, 3 Actions PRs

**`.cursor/rules/dependabot-workflow.mdc`**
- Agent rule for triaging, validating, and merging Dependabot PRs
- CI gate: format:check → lint → tsc → test; Android build + smoke test for risky bumps
- One lockfile merge at a time; final validation on `main` after batches

**`docs/guides/dependabot-process.md`**
- Human/agent process guide for Fluent Mobile (npm, RN 0.84, Android-only)

**`docs/guides/dependabot-resolution-log.md`**
- Empty template for logging batch Dependabot resolutions

**`docs/AGENT_ONBOARDING.md`** / **`.cursor/rules/project-overview.mdc`**
- Links and pointers to Dependabot config and workflow

**Lines changed**: ~441 lines added | **Files modified**: 6

### Dependencies & Configuration
- [x] No new dependencies
- [ ] New dependencies:
- [x] No config changes
- [ ] Environment/build config changes:
- [x] No database changes
- [ ] Database/storage changes:

---

### Testing Completed
- [ ] ✅ Android device/emulator tested (N/A — no app/runtime changes)
- [ ] ✅ Unit tests added/updated and passing (N/A — docs/config only)
- [x] ✅ Edge cases considered and tested (workflow covers RN pin drift, lockfile corruption, native module bumps)

### Code Quality (fluent-mobile gates)
- [ ] ✅ `npm run lint` (N/A — no source changes)
- [ ] ✅ `npm run format:check` (N/A — no source changes)
- [ ] ✅ `npx tsc --noEmit` (N/A — no source changes)
- [ ] ✅ `FLUENT_USER_EMAIL=test@example.com npm test -- --ci` (N/A — no source changes)
- [x] ✅ Self-reviewed the code changes

---

## 📸 Screenshots/Recordings
N/A — infrastructure and documentation only.

---

## 🎯 Why This Solution?
1. **Proven pattern**: Builds on an established Dependabot workflow structure, with project-specific configuration tailored for this repository.
2. **Lowest risk**: Ignores RN line upgrades and pinned React versions; enforces one lockfile merge + `@dependabot rebase` between merges.
3. **Repo-aligned**: Uses npm, `package-lock.json`, existing CI workflow names (Lint Check, Test Check, Build Check), and Fluent-specific smoke-test paths (Projects → Chapters → Verse detail, Sync).

## 📱 Before/After
- **Before**: No Dependabot config; no documented safe merge process for dependency PRs.
- **After**: Weekly Dependabot PRs with grouped updates, guardrails, and Cursor/docs guidance for processing them.

---

## ⚠️ Breaking Changes
- [x] No breaking changes
- [ ] Breaking changes documented below:

---

## 📊 Performance Impact

**Bundle Size:**
- [x] No significant impact (< 1MB)
- [ ] Minor increase (1-5MB)
- [ ] Significant increase (> 5MB) - justified because:

**Runtime Performance:**
- [x] No performance impact
- [ ] Performance improvements
- [ ] Potential performance concern - mitigated by:

**Battery/Memory:**
- [x] No impact on battery or memory usage
- [ ] Optimizations that improve battery/memory
- [ ] Potential impact - acceptable because:

---

## 🧪 How to Test

**Prerequisites:** Node 24+, `npm install`

**Steps:**
1. Review `.github/dependabot.yml` — confirm ignore rules match RN 0.84.1 pins in `package.json`.
2. Confirm `teamgloo/mobile-team` is the correct GitHub reviewer team for this repo (or request a change before merge).
3. Read `docs/guides/dependabot-process.md` and `.cursor/rules/dependabot-workflow.mdc` — verify commands match `.cursor/rules/commands.mdc` and CI workflows.
4. After merge to `main`, confirm Dependabot opens PRs on the next scheduled run (or check repo Dependabot settings / security tab).

**Expected:** No app behavior change. Dependabot becomes active with conservative limits and RN/React guards. Future dependency PRs follow the documented one-at-a-time merge process.

---

## 🔗 Additional Context

1. **Repository-specific cleanup**: Removes unrelated platform, package manager, and messaging references while preserving the relevant reviewer ownership for this project.

GitHub already reports existing vulnerabilities on default branch; this PR enables structured ongoing updates rather than fixing them directly.

**Deployment Notes:** Merge to `main` only — Dependabot reads config from default branch.

**Follow-up Tasks:**
- [x] Confirm `teamgloo/mobile-team` reviewer assignment
- [x] Process first Dependabot PR batch using the new workflow
- [x] Optionally add `typecheck` npm script for parity with agent docs

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      - dependency-name: 'react-test-renderer'
+        update-types:
+          - 'version-update:semver-major'
+          - 'version-update:semver-minor'
+      # Native SQLite — validate binary compatibility before bumping
+      - dependency-name: '@op-engineering/op-sqlite'
+        update-types:
+          - 'version-update:semver-major'
+    reviewers:
+      - 'teamgloo/mobile-team'
```
Comment (kaseywright) [inline .github/dependabot.yml]: I'm not familiar with `teamgloo/mobile-team`. Who will this notify?

Comment (mattrace-gloo) [reply .github/dependabot.yml]: Good catch @kaseywright - teamgloo/mobile-team was copied over from messaging-expo by mistake. I switched it to eten-tech-foundation/fluent-admin, so that group will get notified now.

Happy to use scribe-devs instead if that’s the better team for dependency PRs.

Comment (kaseywright) [reply .github/dependabot.yml]: Yes, that would work!

**Review by kaseywright — APPROVED** [review]
